sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManager.kt

```diff
@@ -166,6 +168,32 @@ interface ViewAnnotationManager {
    */
   val annotations: Map<View, ViewAnnotationOptions>
 
+  /**
+   * Return camera options showing given view annotation list, padding, bearing and pitch values.
+   * Annotations with [ViewAnnotationOptions.visible] set to false will be excluded from the calculations of [CameraOptions].
+   * Annotations with only [View.VISIBLE] will be included in the calculations for [CameraOptions]
+   *
+   * Note: This API isn't supported by Globe projection and will return NULL.
+   * Calling this API immediately after adding the view is a no-op.
+   * Please refer to [OnViewAnnotationUpdatedListener] documentation for understanding the exact moment of time when
+   * view annotation is positioned.
+   *
+   * @param annotations view annotation list to be shown. Annotations should be added beforehand
+   * with [ViewAnnotationManager.addViewAnnotation] API.
+   * @param edgeInsets paddings to apply.
+   * @param bearing camera bearing to apply.
+   * @param pitch camera pitch to apply.
+   *
+   * @return [CameraOptions] object or NULL if [annotations] list is empty.
+   *
+   */
+  fun cameraForAnnotations(
+    annotations: List<View>,
+    edgeInsets: EdgeInsets? = null,
```

Comment:
try to set
`EdgeInsets(-1000000.0, ...` or `100000` and it's break sample. not crash, just blank screen

Reply:
I haven't seen any validation on `EdgeInsets`. if users provide really large edgeInsets, imo its their responsibility. I can however check how does `cameraForCoordinate` or `coordinateBoundsForCamera` works in case of these values.

Reply:
gl-native log error when it couldn't calculate correct zoom, although it still provides wrong cameraOptions. so its upto users to provide correct edgeinsets or they will have errors in log.